src/core/others/ojph_file.cpp

```diff
@@ -191,4 +198,92 @@ namespace ojph {
   }
 
 
+  ////////////////////////////////////////////////////////////////////////////
+  //
+  //
+  //
+  //
+  //
+  ////////////////////////////////////////////////////////////////////////////
+
+  ////////////////////////////////////////////////////////////////////////////
+  mem_outfile::mem_outfile() :
+    is_open(false),
+    size(0),
+    data(NULL),
+    cur_ptr(NULL)
+  {
+  }
+  
+  mem_outfile::~mem_outfile()
+  {
+    free(data);
+    this->is_open = false;
+    this->size = 0;
+    this->data = NULL;
+    this->cur_ptr = NULL;
+  }
+
+  void mem_outfile::open(size_t initial_size /* = 65535 */)
+  {
+    assert(this->is_open == false);
+    assert(this->size == 0);
+    assert(this->data == NULL);
+    assert(this->cur_ptr == NULL);
+    
+    // do initial buffer allocation
+    this->is_open = true;
+    this->size = initial_size;
+    this->data = (ui8*)malloc(this->size);
+    this->cur_ptr = this->data;
+  }
+
+  void mem_outfile::close() {
+    assert(this->is_open == true);
+    assert(this->size);
+    assert(this->data);
+    assert(this->cur_ptr);
+
+    this->is_open = false;
+  }
+  
+  void mem_outfile::release_data() 
+  {
+    assert(this->is_open == false);
+    assert(this->size);
+    assert(this->data);
+    assert(this->cur_ptr);
+    
+    this->is_open = false;
+    this->size = 0;
+    this->data = NULL;
+    this->cur_ptr = NULL;
+  } 
+
```

Comment:
You forgot to deallocate memory

Reply:
whoops, do you want a new PR or did you fix in your merge?  <embarrassed>

Reply:
Don't worry.
I removed this function.

Reply:
Wait a sec - this was actually intentional.  It allowed the caller to take ownership of the underling buffer avoiding a copy.  I would much rather use std::shared_ptr<> for this, but since this library is not using the standard c++ library and using c style memory management, this is is necessary.